### PR TITLE
NHWC PR

### PR DESCRIPTION
**Pull request by jichangjichang**
This PR is all implementation of Tensile for convolution NHWC enablement.
fp32, fp16 and bf16 have been verified on platform gfx908.

### CODE REVIEW

**Review by aazz44ss — APPROVED** [review]
LGTM

File: Tensile/KernelWriter.py [hunk 1]
```diff
@@ -2847,8 +2844,8 @@ def initKernel(self, kernel, tensorParametersA, tensorParametersB ):
 
     tensorParametersA["PackBatchDims"] = kernel["PackBatchDims"] if kernel["PackBatchDims"] & 0x1 else 0
     tensorParametersB["PackBatchDims"] = kernel["PackBatchDims"] if kernel["PackBatchDims"] & 0x2 else 0
-    tensorParametersA["PackedIndices"] = kernel["PackedC0IndicesX"]
-    tensorParametersB["PackedIndices"] = kernel["PackedC1IndicesX"]
+    tensorParametersA["PackedIndices"] = kernel["PackedC%uIndicesX"%self.tPA["tile01Idx"]]# if self.tPA["tileIdx"] == 0 else kernel["PackedC1IndicesX"]
+    tensorParametersB["PackedIndices"] = kernel["PackedC%uIndicesX"%self.tPB["tile01Idx"]]# if self.tPB["tileIdx"] != 0 else kernel["PackedC0IndicesX"]
```
Comment (sdquiring) [inline Tensile/KernelWriter.py:2848]: Does this commented-out code need to remain?

File: Tensile/KernelWriter.py [hunk 2]
```diff
@@ -617,8 +617,8 @@ def makeSubIterSchedule(self, kernel, localReadCode, iteration, pointerLWCode, p
           # calculate the data index of this mfma used for A and B
           # if i // kernel["MIWaveTile"][0]==0, mfma will use new A (need to take iu into account)
           # if i % kernel["MIWaveTile"][0]==0, mfma will use new B
-          packAIdx += instPerPack if i//(kernel["MIWaveTile"][0]+kernel["MIWaveTile"][0]*kernel["MIWaveTile"][1]*(i//(kernel["MIWaveTile"][0]*kernel["MIWaveTile"][1]))) == 0 else 0
-          packBIdx += instPerPack if i % kernel["MIWaveTile"][0] == 0 else 0
+          packAIdx += instPerPack if i//(kernel["MIWaveTileA"]+kernel["MIWaveTileA"]*kernel["MIWaveTileB"]*(i//(kernel["MIWaveTileA"]*kernel["MIWaveTileB"]))) == 0 else 0
```
Comment (sdquiring) [inline Tensile/KernelWriter.py:620]: This seems like we're moving in the wrong direction in terms of maintainability.

**Review by sdquiring — APPROVED** [review]

File: Tensile/KernelWriter.py (hunk 1 above)
Comment (jichangjichang) [reply Tensile/KernelWriter.py]: Done.

**Comment by jichangjichang** [comment]
ALL CI pass, will update changlog and then merge

**Pull request merged**